### DISCUSSION TITLE
More options to visualize trail lengths and colors

### DISCUSSION THREAD

**Discussion by olinesn** [talmolab/sleap#1486] [Ideas]
**Problem:** There are many features that could be visualized with the tracks display that can help with proofreading and other applications.

**Desired functionality:**
- Trail visualization options might need to get moved into its own GUI widget instead of a menu option.
- More dynamic range of trail lengths. Currently fixed list of values, but we should instead have a slider or numeric textbox to allow for any (valid >0) number.
- Opacity
- Hue modifiers to the colormap
- Dynamic colormap (e.g., representing velocity to better be able to spot switches)
- "Future" segments of the trail, not just the past frames
- Trail length scaled by video FPS
- Setting the node that's used as the trail reference point (currently uses the first node in the skeleton?)

**Related issue:** #663

-----
_Original issue:_

Hello,

Writing to submit a feature request, which would be to allow more options for trail length (a 25 option), and to see if the trails could be a slightly different color than their tracks to help them stand out relative to the skeletons. I think just making them a little darker would work, like multiplying the RGB values by 75%.

Thanks!

**Comment by olinesn** [comment]
Also - being able to select a different node for the trail would be great too!

**Comment by olinesn** [comment]
Finally, an indicator for the 'speed' would be nice, maybe highly saturated for high speed, and less saturated for less movement? So when the line between two frames is long, it's more saturated, and when the animal pauses in a location, the line is less saturated. This would be a nice visual signal for when a 'switch' happens that needs to be corrected, suddenly the trail at that frame would be saturated.

**Comment by talmo** [comment]
How about something like this?

![image](https://user-images.githubusercontent.com/3187454/141831487-89ad0ae9-23e1-44b3-b468-b0c4b07ce1ae.png)

**Reply by djg9324** [reply]
How would one obtain a visualization like this in their own video frames?

**Comment by olinesn** [comment]
Just to clarify, how do you think this work work with multiple animals? Aside from that, I do like that you can tell units of time by color.

**Comment by olinesn** [comment]
Hi all (@talmo @roomrys ), hope you're doin well! I'm doing a lot of proofreading right now, and a feature like this would really help. It's hard to track the trail as distinct from the skeleton, but the simplest feature that would make a big difference would be to make the trail color be a darker shade of the skeleton color.

Below is an illustration of how I think it would work.

A lower priority feature would be fore the trail to go grey when it's skipping frames. Otherwise it can look like the trail is showing a node with very high velocity if the node is not tracked for a few frames while the animal is moving.

![SLEAP_trails_v03-01](https://user-images.githubusercontent.com/69451171/210154787-f6e1f4e0-109b-4d91-b3db-5b8e867a4d24.png)

**Comment by roomrys** [comment]
@olinesn,

While I'm messing with this bit of code, would it be helpful, hurtful, or neither to fade to a darker color (maybe start at a darker shade and capped to a non-black shade)?

![image](https://user-images.githubusercontent.com/38435167/210452062-f888a5cd-eaf0-4975-b15c-555f2cae44c8.png)

**Comment by olinesn** [comment]
Hi @roomrys ,

Thanks for the option, yes I would use it in the meantime, and others might even prefer it if you offered both in the future!

Stefan
